extras/nh-ad.c

```diff
@@ -0,0 +1,186 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<math.h>
+
+#define VECSIZE 6
+
+int __enzyme_autodiff(void *, ...);
+int enzyme_dup; 
+int enzyme_const;
+// Computes J - volume change
+double get_Jm1(double E[VECSIZE]){
+    // J = |F| = sqrt(|C|) = sqrt(|I_3 + 2*E|)
+    // Calculate C = [a, d, f, e, c, b]
+    double C[6] = {2.*E[0]+1., 2.*E[1]+1., 2.*E[2]+1., 2.*E[3], 2.*E[4], 2.*E[5]};
+
+    // Calculate determinate of C = adf - (ae^2 + dc^2 + fb^2) + 2bce
+    double det_C = C[0]*C[1]*C[3] - (C[0]*C[3]*C[3] + C[1]*C[4]*C[4] + C[2]*C[5]*C[5]) + 2.*C[5]*C[4]*C[3];
+
+    // Calculate and return J
+    double Jm1 = sqrtf(fabs(det_C)) - 1.;
+    return Jm1;
+    
+}
+
+double computeDetCM1(double *E2work) {
+  return E2work[0]*(E2work[1]*E2work[2]-E2work[3]*E2work[3]) +
+         E2work[5]*(E2work[4]*E2work[3]-E2work[5]*E2work[2]) +
+         E2work[4]*(E2work[5]*E2work[3]-E2work[4]*E2work[1]) +
+         E2work[0] + E2work[1] + E2work[2] +
+         E2work[0]*E2work[1] + E2work[0]*E2work[2] +
+         E2work[1]*E2work[2] - E2work[5]*E2work[5] -
+         E2work[4]*E2work[4] - E2work[3]*E2work[3];
+};
+
+int computeMatinvSym(double A[3][3], double detA, double *Ainv) {
+  // Compute A^(-1) : A-Inverse
+  double B[VECSIZE] = {A[1][1]*A[2][2] - A[1][2]*A[2][1], 
+                 A[0][0]*A[2][2] - A[0][2]*A[2][0], 
+                 A[0][0]*A[1][1] - A[0][1]*A[1][0], 
+                 A[0][2]*A[1][0] - A[0][0]*A[1][2], 
+                 A[0][1]*A[1][2] - A[0][2]*A[1][1], 
+                 A[0][2]*A[2][1] - A[0][1]*A[2][2]  
+                };               
+  for ( int m = 0; m < VECSIZE; m++) Ainv[m] = B[m] / (detA);
+  return 0;
+};
+
+double log1p_series_shifted(double x) {
+  double left = sqrt(2.)/2 - 1;
+  double right = sqrt(2.) - 1;
+  double sum = 0;
+  if (x < left) {
+      sum -= log(2.) / 2;
+      x = 1 + 2 * x;
+  } else if (right < x) {
+      sum += log(2.) / 2;
+      x = (x - 1) / 2;
+  }
+  double y = x / (2. + x);
+  double y2 = y*y;
+  sum += y;
+  y *= y2;
+  sum += y / 3;
+  y *= y2;
+  sum += y / 5;
+  y *= y2;
+  sum += y / 7;
+  return 2 * sum;
+};
+
+/* Function that defines the Strain energy. Currently unsing the neo-hookean model, but would like
+user input for this function*/
+void psi_func(double *psi, double E[VECSIZE], double lambda, double mu){
+
+/* 
+Inputs
+---------------------------------------------------------------------
+E      - 
+lambda - Lame's first parameter 
+mu     - Shear modulus
+J      - Determintate of Deformation (Volume Change)
+
+Output
+---------------------------------------------------------------------
+psi - Elastic Stain Energy
+*/
+
+double Jm1 = get_Jm1(E);
+
+// psi = lambda/2*log(J)^2 - mu*log(J) + mu*trace(E)
+*psi = lambda/2.*log1p_series_shifted(Jm1)*log1p_series_shifted(Jm1) - mu*log1p_series_shifted(Jm1) + mu*(E[0] + E[1] + E[2]); 
+}
+
+// Computes S via eq 15(St. Venant-Kirchoff model) to check S computed by enzyme
```

Comment:
I guess this is a reference to https://ratel.micromorph.org/doc/modeling/materials/neo-hookean/?highlight=venant#equation-eq-st-venant-kirchoff
but actually implementing https://ratel.micromorph.org/doc/modeling/materials/neo-hookean/?highlight=venant#equation-neo-hookean-stress-stable

Reply:
Good point! I don't think this test is verified. I think we need to remove this file for this PR and only focus on `nh-ratel-test.c`.

Reply:
@LeilaGhaffari You're right, I can remove this file from the PR.